### PR TITLE
Create tenant fetcher secret only for local setup

### PR DESCRIPTION
**Pull request by StefanCholakov**
**Description**

Changes proposed in this pull request:
- Add branching to create tenant fetcher secret only on local setup

**Pull Request status**

- [x] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

**Review by desislavaa — APPROVED** [review]

**Review by nyordanoff — DISMISSED** [review]

**Review by alextargov — DISMISSED** [review]

File: installation/resources/installer-config-gke-benchmark.yaml.tpl [hunk 1]
```diff
@@ -31,8 +31,18 @@ data:
...
\ No newline at end of file
+  global.cockpit.auth.path: "/jwks"
+  global.tenantFetcher.endpoints.subaccountCreated: "http://compass-external-services-mock.compass-system.svc.cluster.local:8080/tenant-fetcher/subaccount-create"
+  global.tenantFetcher.secret.name: "compass-tenant-fetcher-secret"
+  global.tenantFetcher.secret.clientIdKey: "client-id"
+  global.tenantFetcher.secret.oauthMode: "oauth-mtls"
+  global.tenantFetcher.secret.clientCertKey: "client-cert"
+  global.tenantFetcher.secret.clientKeyKey: "client-key"
+  global.tenantFetcher.secret.oauthUrlKey: "url"
+  global.tenantFetcher.secret.skipSSLValidation: "true"
```
Comment (StanislavStefanov) [inline installation/resources/installer-config-gke-benchmark.yaml.tpl:45]: We have the **exact same** values in installer-config-local.yaml.tpl and in Value.yaml. Why do we replicate them? It should be enough for these values to be maintained only in Values.yaml. Also in Values.yaml the values are not quoted which may be a problem.

**Review by StanislavStefanov — CHANGES_REQUESTED** [review]

File: installation/resources/installer-config-gke-benchmark.yaml.tpl (hunk 1 above)
Comment (StanislavStefanov) [reply installation/resources/installer-config-gke-benchmark.yaml.tpl:45]: Screen Shot 2022-04-21 at 14 27 34

File: chart/compass/values.yaml [hunk 2]
```diff
@@ -494,17 +495,17 @@ global:
       consumerSubaccountIdsLabelKey: "consumer_subaccount_ids"
     containerName: "tenant-fetcher"
     oauth:
-      client: ""
+      client: """client_id"
```
Comment (StanislavStefanov) [inline chart/compass/values.yaml:498]: Remove additional "" in the beginnig

File: chart/compass/values.yaml [hunk 3]
```diff
@@ -494,17 +495,17 @@ global:
...
       secret: ""
-      tokenURL: ""
-      tokenPath: ""
+      tokenURL: ""'{{ printf "https://%s.%s" .Values.global.externalServicesMock.certSecuredHost .Values.global.ingress.domainName }}'
```
Comment (StanislavStefanov) [inline chart/compass/values.yaml:500]: Remove additional "" in the beginnig

File: chart/compass/values.yaml [hunk 4]
```diff
@@ -494,17 +495,17 @@ global:
...
+      tokenPath: """/cert/token"
```
Comment (StanislavStefanov) [inline chart/compass/values.yaml:501]: Remove additional "" in the beginnig

**Review by StanislavStefanov — CHANGES_REQUESTED** [review]

**Review by StanislavStefanov — APPROVED** [review]

**Pull request merged**